tests/bluetooth/ctrl_isoal/src/sub_sets/isoal_test_tx.c

```diff
@@ -4396,8 +4396,501 @@ ZTEST(test_tx_framed, test_tx_framed_2_sdu_3_frag_4_pdu)
  */
 ZTEST(test_tx_framed, test_tx_framed_2_sdu_3_frag_4_pdu_padding)
 {
-	PRINT("Framed padding not implemented.\n");
-	ztest_test_skip();
+	uint8_t testdata[(TEST_TX_PDU_PAYLOAD_MAX * 2) -
+		((PDU_ISO_SEG_HDR_SIZE * 2) + PDU_ISO_SEG_TIMEOFFSET_SIZE)];
+	struct tx_pdu_meta_buffer tx_pdu_meta_buf[2];
+	struct tx_sdu_frag_buffer tx_sdu_frag_buf;
+	struct isoal_pdu_buffer pdu_buffer[2];
+	struct pdu_iso_sdu_sh seg_hdr[2 * 2];
+	isoal_source_handle_t source_hdl;
+	isoal_sdu_len_t sdu_total_size;
+	isoal_pdu_len_t pdu_write_size;
+	uint32_t stream_sync_delay;
+	uint64_t sdu_packet_number;
+	uint32_t group_sync_delay;
+	uint8_t iso_interval_int;
+	uint64_t payload_number;
+	uint32_t sdu_timestamp;
+	uint16_t testdata_indx;
+	uint16_t testdata_size;
+	uint16_t pdu_write_loc;
+	uint16_t sdu_read_loc;
+	uint64_t event_number;
+	uint32_t sdu_interval;
+	uint8_t sdu_fragments;
+	uint16_t pdu_hdr_loc;
+	uint32_t ref_point;
+	isoal_status_t err;
+	uint8_t max_octets;
+	uint8_t role;
+	uint8_t BN;
+	uint8_t FT;
+
+	/* Settings */
+	role = BT_CONN_ROLE_PERIPHERAL;
+	iso_interval_int = 2;
+	sdu_interval = CONN_INT_UNIT_US + 50;
+	max_octets = TEST_TX_PDU_PAYLOAD_MAX + 5;
+	BN = 6;
+	FT = 1;
+	stream_sync_delay = (iso_interval_int * CONN_INT_UNIT_US) - 200;
+	group_sync_delay = (iso_interval_int * CONN_INT_UNIT_US) - 50;
+
+	/* SDU 1 Frag 1 ------------------------------------------------------*/
+	isoal_test_init_tx_pdu_buffer(&tx_pdu_meta_buf[0]);
+	isoal_test_init_tx_pdu_buffer(&tx_pdu_meta_buf[1]);
+	isoal_test_init_tx_sdu_buffer(&tx_sdu_frag_buf);
+	init_test_data_buffer(testdata,
+		(TEST_TX_PDU_PAYLOAD_MAX * 2) -
+		((PDU_ISO_SEG_HDR_SIZE * 2) + PDU_ISO_SEG_TIMEOFFSET_SIZE));
+	(void)memset(&seg_hdr, 0, sizeof(seg_hdr));
+	pdu_buffer[0].handle = (void *)&tx_pdu_meta_buf[0].node_tx;
+	pdu_buffer[0].pdu = (struct pdu_iso *)tx_pdu_meta_buf[0].node_tx.pdu;
+	pdu_buffer[0].size = TEST_TX_PDU_PAYLOAD_MAX;
+	pdu_buffer[1].handle = (void *)&tx_pdu_meta_buf[1].node_tx;
+	pdu_buffer[1].pdu = (struct pdu_iso *)tx_pdu_meta_buf[1].node_tx.pdu;
+	pdu_buffer[1].size = TEST_TX_PDU_PAYLOAD_MAX;
+	sdu_packet_number = 2000;
+	event_number = 2000;
+	sdu_timestamp = 9249;
+	ref_point = 9249 + (iso_interval_int * CONN_INT_UNIT_US) - 50;
+	sdu_total_size = (TEST_TX_PDU_PAYLOAD_MAX * 2) -
+		((PDU_ISO_SEG_HDR_SIZE * 2) + PDU_ISO_SEG_TIMEOFFSET_SIZE);
+	testdata_indx = 0;
+	testdata_size = ((TEST_TX_PDU_PAYLOAD_MAX * 2) -
+		((PDU_ISO_SEG_HDR_SIZE * 2) + PDU_ISO_SEG_TIMEOFFSET_SIZE))/3;
```

Comment:
Several more cases of `2`: If they all represent the same value, then I strongly suggest to define that as a `#define` or a `const` value, and use that, so that it is easier to maintain.

Reply:
I'll check if a const is an option. If recall correct using that in an array declaration is c version dependent.

Reply:
> c version dependent.

If it is, it's definitely allowed in C11 which is C version Zephyr uses. I think most issues around this is using a variable length declaration, which wouldn't be the case with a `const`. 

If it is a problem, then you can use a `#define` :)

Reply:
Done.